sentry_sdk/integrations/asgi.py

```diff
@@ -267,13 +267,19 @@ def event_processor(self, event, hint, asgi_scope):
         event["request"] = deepcopy(request_data)
 
         # Only set transaction name if not already set by Starlette or FastAPI (or other frameworks)
-        already_set = event["transaction"] != _DEFAULT_TRANSACTION_NAME and event[
-            "transaction_info"
-        ].get("source") in [
-            TransactionSource.COMPONENT,
-            TransactionSource.ROUTE,
-            TransactionSource.CUSTOM,
-        ]
+        transaction = event.get("transaction")
+        transaction_source = event.get("transaction_info", {}).get("source")
```

Comment:
We should be more defensive here: this can fail if `transaction_info` is on the `event` but is `None`.

```suggestion
        transaction_source = (event.get("transaction_info") or {}).get("source")
```